projects/igniteui-angular/src/lib/calendar/calendar-base.ts

```diff
@@ -649,8 +649,11 @@ export class IgxCalendarBaseDirective implements ControlValueAccessor {
      * @hidden
      */
     private selectSingle(value: Date) {
-        this.selectedDates = this.getDateOnly(value);
-        this._onChangeCallback(this.selectedDates);
+        const selectedDate = this.selectedDates as Date;
+        if (!selectedDate || selectedDate.getTime() !== value.getTime()) {
```

Comment:
This cast is not correct. The fact that the next line checks for `null/undefined` means that it should have been at least `Date | null | undefined`, but what worries me more is if it's not a Date object the `.getTime()` in the next line will crash.

I think adding a compare method will be safer:
```ts
private selectionEquals(a: unknown, b: unknown): boolean {
   if (a instanceof Date && b instanceof Date) {
       return a.getTime() === b.getTime();
    } else if (Array.isArray(a) && Array.isArray(b)) {
        return a.length === b.length && a.every((v, i) => this.selectionEquals(v, b[i]));
    } else {
        return a === b;
    }
}
```

Also, by looking at the code, it's very probable that the same issue can be reproduced with arrays in multiple selection mode